batcher/batcher.mligo

```diff
@@ -1692,6 +1732,94 @@ let get_valid_swap_reduced
  | Some vswp -> vswp
  | None -> failwith swap_does_not_exist
 
+[@inline]
+let refund_orders
+  (refund_address: address)
+  (ots: ordertypes)
+  (valid_swap:valid_swap)
+  (storage: storage): result =
+  let fee = storage.fee_in_mutez in
+  let collect_refunds ((tam,mutez_to_ref), (ot, amt): ((token_amount_map * tez) * (ordertype * nat))) : (token_amount_map * tez) =
+    let token  = match ot.side with
+                 | Buy -> valid_swap.swap.from.token
+                 | Sell -> valid_swap.swap.to
+    in
+    let ta = {
+       token = token;
+       amount = amt;
+    } in
+    let tam = TokenAmountMap.increase ta tam in
+    let mutez_to_ref = mutez_to_ref + fee in
+    tam, mutez_to_ref
+  in
+  let token_refunds, tez_refunds= Map.fold collect_refunds ots ((Map.empty: token_amount_map), 0mutez) in
+  let treasury_vault = Treasury.get_treasury_vault () in
+  let operations = Treasury_Utils.transfer_holdings treasury_vault refund_address token_refunds in
+  let operations = if tez_refunds > 0mutez then Treasury_Utils.transfer_fee refund_address tez_refunds :: operations else operations in
```

Comment:
I don't think we should refund users who cancel their orders. We're still paying for the oracle update for this batch, and other people may have added their order to match the one that got cancelled.

Reply:
This would make it inconsistent with redemptions where people weren't filled.  @oteku Do you have a deciding view?